src/anyio/_backends/_asyncio.py

```diff
@@ -190,11 +189,15 @@ def __init__(self, deadline: float = math.inf, shield: bool = False):
         self._host_task: Optional[asyncio.Task] = None
         self._timeout_expired = False
 
+    def _timeout(self):
+        self._timeout_expired = True
```

Comment:
I think this should check if the scope has already been cancelled. In such cases we don't want to set the `_timeout_expired` flag. In a later PR we should also check if the deadline has actually passed, since trio at least makes it possible to alter the deadline.

Reply:
do you have a test case for that change?

Reply:
Probably not since the current code would not have done the right thing either. I'll take a look at the tests anyway.